### PR TITLE
fix: respect sudo.enable setting when running log plugins

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Fix a bug where the log plugin was not respecting the `sudo.enable` setting, so it would always run the plugins using sudo.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s. You can activate automatic signing by running `just prepare-dev` once)
- [x] I ran `just format` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `just check` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot [hunk 1]
```diff
@@ -232,6 +232,29 @@ Reporting sudo misconfiguration
     ...    current_only=${True}
     ...    text=ERROR log plugins: Skipping /etc/share/tedge/log-plugins/dummy_plugin: not properly configured to run with sudo
 
+Supported log types loaded when sudo disabled
+    [Documentation]    Log plugins should be discovered and listed even when sudo.enable is false.
+    ...    Verifies that the agent does not unconditionally invoked sudo during plugin detection.
+    [Tags]    adapter:docker
+    Sleep    1s    reason=Allow the agent to complete startup actions before stopping
+    Stop Service    tedge-agent
+
```
Comment (rina23q) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:241]: Probably better to assert the check if `tedge-agent` is really stopped?
```suggestion
    Service Health Status Should Be Down    tedge-agent

```

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot [hunk 2]
```diff
@@ -232,6 +232,29 @@ Reporting sudo misconfiguration
...
+    # remove tedge sudo rules to prevent any command from working
+    Execute Command    cmd=rm -f /etc/sudoers.d/tedge
+
+    Execute Command    tedge config set sudo.enable false
+
+    # install a new plugin
+    Execute Command    mkdir -p /etc/share/tedge/log-plugins
```
Comment (rina23q) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot]: Mistake using `/etc` instead of `/usr`?

```suggestion
    Execute Command    mkdir -p /usr/share/tedge/log-plugins
```

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot (hunk 1 above)
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:241]: Though the command will fail if the tedge-agent stopped (as it is using systemctl under the hood), so a check isn't needed (as we don't need to check that systemctl did what it says it will do).

Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:241]: Though the 1s sleep hack is debatable as I think we have a problem in some actors which result in a slow shutdown (~60s), so I figured if the extra second helps us avoid a 60s timeout, then it is worthwhile using until we address the underlying reason

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot (hunk 2 above)
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot]: It was just copied from another test, but yeah you're right that using a custom folder isn't really adding anything to the test so I'll remove that part and just use the default folder.

Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot]: Resolved in d2b3cbd2c9d00d9a1253602a69ff753346b98383 where I just removed the folder creation and just used the default plugin directory

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot (hunk 1 above)
Comment (rina23q) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:241]: > Though the 1s sleep hack is debatable as I think we have a problem in some actors which result in a slow shutdown (~60s),

Yeah, exactly my point is that, if the corner case occurs (tedge-agent doesn't shut down within 60s), then the keyword can tell at least the failure reason clearly.

Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:241]: But in both cases the tedge-agent will be stopped after the "Stop Service" keyword returns successfully

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot [hunk 3]
```diff
@@ -232,6 +232,27 @@ Reporting sudo misconfiguration
     ...    current_only=${True}
     ...    text=ERROR log plugins: Skipping /etc/share/tedge/log-plugins/dummy_plugin: not properly configured to run with sudo
 
+Supported log types loaded when sudo disabled
+    [Documentation]    Log plugins should be discovered and listed even when sudo.enable is false.
+    ...    Verifies that the agent does not unconditionally invoked sudo during plugin detection.
+    [Tags]    adapter:docker
```
Comment (albinsuresh) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:238]: Out of curiosity, any specific reason why this tag is applied specifically for this test, unlike the other tests where it is added to `Test Tags` under `Settings`?

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot [hunk 4]
```diff
@@ -232,6 +232,27 @@ Reporting sudo misconfiguration
...
+    Sleep    1s    reason=Allow the agent to complete startup actions before stopping
```
Comment (albinsuresh) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:239]: Wouldn't that be enough?
```suggestion
    Service Health Status Should Be Up    tedge-agent
```

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot (hunk 3 above)
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:238]: The reason was because it is doing a destructive action (removing the sudoers rule). Since we don't have a nice interface to revert this action in all cases (without overriding the automated cleanup), then it would break another test environment if we were to use the ssh adapter etc.

File: tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot (hunk 4 above)
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_plugins.robot:239]: Unfortunately the health status doesn't tell us if the agent is actually doing any work or not.
